File: app/src/main/java/com/example/java/android1/java_android_notes/ui/NoteDescriptionFragment.java
package com.example.java.android1.java_android_notes.ui;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.java.android1.java_android_notes.MainActivity;
import com.example.java.android1.java_android_notes.R;
import com.example.java.android1.java_android_notes.data.DataNote;
import com.example.java.android1.java_android_notes.data.DataNoteSource;
import com.example.java.android1.java_android_notes.data.DataNoteSourceFirebase;
import com.example.java.android1.java_android_notes.dialogs.DialogConfirmFragment;
import com.example.java.android1.java_android_notes.service.Navigation;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

public class NoteDescriptionFragment extends Fragment {

    public static final String ARG_NOTE = "NoteDescriptionFragment.note";
    private static final String KEY_DIALOG_CONFIRM_DELETE = "ListOfNotesFragment.Dialog.ConfirmDelete";

    private DataNote mDataNote;
    private DataNoteSource mDataNoteSource;

    private int mItemIndex = -1;

    public static NoteDescriptionFragment newInstance(int itemIndex) {
        NoteDescriptionFragment fragment = new NoteDescriptionFragment();
        Bundle args = new Bundle();
        args.putInt(ARG_NOTE, itemIndex);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mItemIndex = getArguments().getInt(ARG_NOTE, -1);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_note_description, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        if (savedInstanceState != null) {
            mItemIndex = savedInstanceState.getInt(ARG_NOTE, -1);
        }
        setHasOptionsMenu(true);
        mDataNoteSource = DataNoteSourceFirebase.getInstance();
        if (mItemIndex == -1 || mItemIndex >= mDataNoteSource.getDataNoteCount()) {
            requireActivity().getSupportFragmentManager().popBackStack();
            return;
        }
        mDataNote = mDataNoteSource.getItem(mItemIndex);
        if (mDataNote != null) {
            initNote(view);
            editNote(view);
        }
    }

    @Override
    public void onCreateOptionsMenu(@NonNull Menu menu, @NonNull MenuInflater inflater) {
        inflater.inflate(R.menu.note_menu_fragment, menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == R.id.action_remove_note) {
            new DialogConfirmFragment(mItemIndex).show(requireActivity().getSupportFragmentManager(), KEY_DIALOG_CONFIRM_DELETE);
        } else {
            return super.onOptionsItemSelected(item);
        }
        return true;
    }

    private void initNote(View view) {
        TextView noteName = generateTextView(view, R.id.item_title, mDataNote.getNoteName());
        TextView noteDescription = generateTextView(view, R.id.item_text, mDataNote.getNoteDescription());
        TextView noteDate = generateTextView(view, R.id.item_date, mDataNote.getNoteDate());
    }

    private TextView generateTextView(View view, int elementId, String text) {
        TextView textView = view.findViewById(elementId);
        textView.setText(text);
        return textView;
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putInt(ARG_NOTE, mItemIndex);
    }

    private void editNote(View view) {
        FloatingActionButton actionButton = view.findViewById(R.id.edit_note);
        actionButton.setOnClickListener((click) -> {
            Navigation navigation = new Navigation(requireActivity().getSupportFragmentManager(), (MainActivity) requireActivity());
            navigation.addFragment(EditNoteFragment.newInstance(mItemIndex), true);
        });
    }

}

File: app/src/main/java/com/example/java/android1/java_android_notes/ui/NoteDescriptionFragment.kt
package com.example.java.android1.java_android_notes.ui

import android.os.Bundle
import android.view.*
import androidx.fragment.app.Fragment
import com.example.java.android1.java_android_notes.MainActivity
import com.example.java.android1.java_android_notes.R
import com.example.java.android1.java_android_notes.data.DataNote
import com.example.java.android1.java_android_notes.data.DataNoteSource
import com.example.java.android1.java_android_notes.data.DataNoteSourceFirebase
import com.example.java.android1.java_android_notes.databinding.FragmentNoteDescriptionBinding
import com.example.java.android1.java_android_notes.dialogs.DialogConfirmFragment
import com.example.java.android1.java_android_notes.service.Navigation

class NoteDescriptionFragment : Fragment() {
    private var _binding: FragmentNoteDescriptionBinding? = null
    private val binding get() = _binding!!

    private var mDataNote: DataNote? = null
    private var mDataNoteSource: DataNoteSource? = null
    private var mItemIndex = -1
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            mItemIndex = requireArguments().getInt(ARG_NOTE, -1)
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentNoteDescriptionBinding.inflate(inflater)
        if (savedInstanceState != null) {
            mItemIndex = savedInstanceState.getInt(ARG_NOTE, -1)
        }
        setHasOptionsMenu(true)
        mDataNoteSource = DataNoteSourceFirebase.instance
        mDataNoteSource?.let {
            if (mItemIndex == -1 || mItemIndex >= it.dataNoteCount) {
                requireActivity().supportFragmentManager.popBackStack()
            } else {
                mDataNote = mDataNoteSource?.getItem(mItemIndex)
                if (mDataNote != null) {
                    initNote()
                    editNote()
                }
            }
        }
        return binding.root
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        inflater.inflate(R.menu.note_menu_fragment, menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.action_remove_note) {
            DialogConfirmFragment(mItemIndex).show(
                requireActivity().supportFragmentManager,
                KEY_DIALOG_CONFIRM_DELETE
            )
        } else {
            return super.onOptionsItemSelected(item)
        }
        return true
    }

    private fun initNote() {
        mDataNote?.apply {
            binding.itemTitle.text = noteName
            binding.itemText.text = noteDescription
            binding.itemDate.text = noteDate
        }
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putInt(ARG_NOTE, mItemIndex)
    }

    private fun editNote() {
        binding.editNote.setOnClickListener {
            val navigation = Navigation(
                requireActivity().supportFragmentManager,
                (requireActivity() as MainActivity)
            )
            navigation.addFragment(EditNoteFragment.newInstance(mItemIndex), true)
        }
    }

    companion object {
        const val ARG_NOTE = "NoteDescriptionFragment.note"
        private const val KEY_DIALOG_CONFIRM_DELETE = "ListOfNotesFragment.Dialog.ConfirmDelete"

        @JvmStatic
        fun newInstance(itemIndex: Int) =
            NoteDescriptionFragment().apply {
                arguments = Bundle().apply {
                    putInt(ARG_NOTE, itemIndex)
                }
            }
    }
}